Props/Cat.gd

```diff
@@ -0,0 +1,34 @@
+extends CharacterBody2D
+
+const speed = 50
+
+var destination: Node2D
+@onready var nav_agent := $NavigationAgent2D as NavigationAgent2D
```

Comment:
The NavigationAgent is able to compute the shortest path to a destination, which we set with the `target_position` param down below.